fitbenchmarking/fitting/controllers/minuit_controller.py

```diff
@@ -0,0 +1,72 @@
+"""
+Implements a controller for the CERN pacakage Minuit
+https://seal.web.cern.ch/seal/snapshot/work-packages/mathlibs/minuit/
+using the iminuit python interface
+http://iminuit.readthedocs.org
+"""
+from iminuit import Minuit
+import numpy as np
+
+from fitbenchmarking.fitting.controllers.base_controller import Controller
+
+class MinuitController(Controller):
+    """
+    Controller for the Minuit fitting software
+    """
+
+    def __init__(self, problem, use_errors):
+        """
+        Initializes variable used for temporary storage.
+        """
+        super(MinuitController, self).__init__(problem, use_errors)
+        self._popt = None
+        self._initial_step = None
+
+    def setup(self):
+        """
+        Setup for Minuit
+        """        
+        # minuit requires an initial step size.
+        # The docs say
+        # "A good guess is a fraction of the initial 
+        #  fit parameter value, e.g. 10% 
+        #  (be careful when applying this rule-of-thumb
+        #  when the initial parameter value is zero "
+        self._initial_step = 0.1 * np.array(self.initial_params)
+        # set small steps to something sensible(?)
+        self._initial_step[self._initial_step < 1e-12] = 1e-12
+
+    def _prediction_error(self, p):
+        f = self.problem.eval_f(x=self.data_x,
+                                params=p,
+                                function_id=self.function_id)
+        f = f - self.data_y
+        if self.use_errors:
+            f = f / self.data_e
+
+        return np.dot(f,f)
+
+    def fit(self):
+        """
+        Run problem with Minuit
+        """
+        self.success = False
+        m = Minuit.from_array_func(self._prediction_error,
```

Comment:
Does this make sense to put in the `setup` function so we just time `m.migrad()`?

Reply:
Fixed.  I also moved the bit that saves as a scipy array into `cleanup`